flink-ml-benchmark/src/main/java/org/apache/flink/ml/benchmark/BenchmarkUtils.java

```diff
@@ -99,6 +99,7 @@ private static BenchmarkResult runBenchmark(
             DataGenerator<?> modelDataGenerator)
             throws Exception {
         StreamExecutionEnvironment env = TableUtils.getExecutionEnvironment(tEnv);
+        env.getConfig().enableObjectReuse();
```

Comment:
According to [1], `enableObjectReuse` can lead to bugs when the user-code function of an operation is not aware of this behavior. So we can avoid bugs if users take care when writing UDF.

If we don't enable object re-use here, users won't be able to reproduce the best possible performance by running our benchmarks, which weakens the purpose of adding these benchmarks.

I am not sure the two points mentioned above are valid concerns:
- We should be able to make sure that all Flink ML algorithms can run with object-reuse enabled. And it is useful to catch and fix issues by running benchmarks with this config.
- User's production code typically won't use BenchmarkUtils. Thus their production code won't be affected by the `enableObjectReuse()` added here.
- While Flink users can change configuration parameters, it is generally preferred to provide the best set of configuration parameters in our benchmark setup with our best effort.

[1] https://nightlies.apache.org/flink/flink-docs-master/docs/dev/datastream/execution/execution_configuration/

Reply:
@zhipeng93 Configuration of the execution environment typically happens in the method where it is instantiated so that other developers can easily find all the env configurations used by the job execution.

How about we move this line to `Benchmark::executeBenchmarks()` to improve readability?

Reply:
Thanks for the suggestion. I agree that users won't use BenchmarkUtils in the production code so it won't lead to wrong results. 

I also agree that we should make sure that FlinkML algs can run with object-reuse enabled. We could add the check in another PR. 

I have moved the setting to `Benchmark::executeBenchmarks()`.

Reply:
Got it. If we can make that Flink ML algorithms won't throw exceptions when object reuse is enabled, then I agree that we can add this configuration to BenchmarkUtils at it does not have other side effects.